website/static/js/addProjectPlugin.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * UI and function to add project
+ */
+'use strict';
+
+require('css/add-project-plugin.css');
+var $ = require('jquery');
+var m = require('mithril');
+var $osf = require('js/osfHelpers');
+
+// XHR configuration to get apiserver connection to work
+var xhrconfig = function (xhr) {
+    xhr.withCredentials = true;
+};
+
+
+var AddProject = {
+    controller : function (options) {
+        var self = this;
+        self.defaults = {
+            buttonTemplate : m('.btn.btn-primary[data-toggle="modal"][data-target="#addProjectModal"]', 'Add new Project'),
+            parentID : null,
+            modalID : 'addProjectModal',
+            stayCallback :null // Function to call when user decides to stay after project creation
+        };
+        self.viewState = m.prop('form'); // 'processing', 'success', 'error';
+        self.options = $.extend(self.defaults, options);
+        self.showMore = m.prop(false);
+        self.newProjectName = m.prop('');
+        self.newProjectDesc = m.prop('');
+        self.newProjectCategory = m.prop('project');
+        self.goToProjectLink = m.prop('');
+        self.saveResult = m.prop({});
+        self.errorMessageType = m.prop('unknown');
+        self.errorMessage = {
+            'unknown' : 'There was an unknown error. Please try again later.'
+        };
+        // Load Category list from API
+        self.categoryList = [];
+        self.loadCategories = function _loadCategories () {
+            m.request({method : 'OPTIONS', url : $osf.apiV2Url('nodes/', { query : {}}), config : xhrconfig}).then(function _success(results){
+                console.log(results);
+                if(results.actions.POST.category){
+                    self.categoryList = results.actions.POST.category.choices;
+                    self.categoryList.sort(function(a, b){ // Quick alphabetical sorting
+                        if(a.value < b.value) return -1;
+                        if(a.value > b.value) return 1;
+                        return 0;
+                    });
+                }
+            }, function _error(results){
+                console.error('Error loading category names:', results);
+            });
+        };
+        self.loadCategories();
+        // Validation
+        self.isValid = m.prop(false);
+        self.checkValid = function _checkValid() {
+            var projectNameNotEmpty = self.newProjectName().trim().length > 0 ? true : false;
```

Comment:
You should be able to remove `isValid`, or at least turn it into a method rather than a prop, since it's derived data.

```js
self.isValid = function() {
    self.newProjectName().trim().length > 0;
}
```